SimG4Core/Application/plugins/OscarMTProducer.cc

```diff
@@ -71,6 +78,24 @@ OscarMTProducer::OscarMTProducer(edm::ParameterSet const& p, const OscarMTMaster
   m_runManagerWorker = std::make_unique<RunManagerMTWorker>(p, consumesCollector());
   m_masterThread = ms;
 
+  // Prepare tokens
+  bool isDD4Hep = p.getParameter<bool>("g4GeometryDD4hepSource");
+  if (!m_hasToken) {
+    G4MUTEXLOCK(&m_OscarMutex);
```

Comment:
There is no need for this lock since we do not run any module constructors concurrently.

Reply:
OK, will remove lock/unlock.

Reply:
Done